llvm/test/CodeGen/X86/O3-pipeline.ll

```diff
@@ -128,6 +128,8 @@
 ; CHECK-NEXT:       Machine Loop Invariant Code Motion
 ; CHECK-NEXT:       Bundle Machine CFG Edges
 ; CHECK-NEXT:       X86 FP Stackifier
+; CHECK-NEXT:       Machine Dominance Frontier Construction
+; CHECK-NEXT:       X86 Load Value Injection (LVI) Load Hardening
```

Comment:
According to @ctopper, since this backport doesn't add the “MachineDominator Tree Construction” pass here (unlike e97a3e5d9d428c4d455fa1b1982728bb71f0c397), the backport may not suffer the same performance issue that de92dc2850c17259090ccf644b2f2375ab8e1664 was designed to fix.

Reply:
Yes, you are correct that the O3 pipelines is mostly unaffected by this change on LLVM 9. However, O3 was never the problematic case in the first place. The problematic case are the changes to the O0 pipeline, which should never perform machine domtree calculations.